pkg/apis/workflow/v1alpha1/data_types.go

```diff
@@ -0,0 +1,59 @@
+package v1alpha1
+
+// Data is a data template
+type Data struct {
+	PodPolicy PodPolicy `json:"podPolicy,omitempty" protobuf:"bytes,1,opt,name=podPolicy,casttype=PodPolicy"`
+
+	// Source sources external data into a data template
+	Source *DataSource `json:"source,omitempty" protobuf:"bytes,2,opt,name=source"`
```

Comment:
I think this isn't optional? if so, no need for pointer or `omitempty`

Reply:
Done